snakemake-mode.el

```diff
@@ -325,6 +328,57 @@ label."
     (nreverse index)))
 
 
+;;; Embedded language syntax-highlighting
+
+(defun snakemake-mode-setup-mmm ()
+  "Set up MMM mode to highlight embedded R code.
+
+You must have the R-strings either within a R(''' ''') function call or a code
+block delimited with '''#r and ''' (triple double-quotes also accepted).
+
+For automatic highlighting of embedded regions, you need to set mmm-global-mode
+to 'maybe in your setup file."
+
+  (progn
+    (unless (featurep 'mmm-mode)
+      (error "You need to install mmm-mode"))
+
+    (when (= mmm-global-mode nil)
```

Comment:
This will fail with a type error.  Could use `(unless mmm-global-mode ...)` instead.

Also, this should be moved after  the `require` call in case someone
is setting `mmm-global-mode` from within an `eval-after-load`.